### PR TITLE
2018 edition

### PR DESCRIPTION
**Pull request by octronics**
Convert this crate to the rust 2018 edition. This is a mostly automated/guided conversion. Check the commit message for details.

### CODE REVIEW

File: src/node/element/mod.rs [hunk 1]
```diff
@@ -1,8 +1,13 @@
 //! The element nodes.
 
+// SVG, Script, Style element has no Default.
```
Comment (IvanUkhov) [inline src/node/element/mod.rs]: Let’s remove comments here and below.

File: src/node/element/mod.rs [hunk 2]
```diff
@@ -229,19 +240,19 @@ implement! {
...
-        inner.append(::node::Text::new(content));
+        inner.append(crate::node::Text::new(content));
     }
 }
 
 #[cfg(test)]
 mod tests {
-    use node::Node;
     use super::{Element, Style};
+    use crate::node::Node;
```
Comment (IvanUkhov) [inline src/node/element/mod.rs:253]: Is it by rustfmt?

File: src/node/element/path/data.rs [hunk 3]
```diff
@@ -1,8 +1,8 @@
 use std::ops::Deref;
 
-use node::Value;
-use parser::{Error, Reader, Result};
 use super::{Command, Number, Parameters, Position};
+use crate::node::Value;
+use crate::parser::{Error, Reader, Result};
```
Comment (IvanUkhov) [inline src/node/element/path/data.rs:5]: Is it by rustfmt?

File: src/parser/reader.rs [hunk 4]
```diff
@@ -113,36 +115,37 @@ impl<'l> Reader<'l> {
...
-            '-'
-            | '.'
-            | '0'...'9'
-            | '\u{B7}'
-            | '\u{0300}'...'\u{036F}'
-            | '\u{203F}'...'\u{2040}' => true,
-            _ => false,
-        })
+        self.consume_name_start_char()
+            || self.consume_if(|c| match c {
```
Comment (IvanUkhov) [inline src/parser/reader.rs:119]: Is it by rustfmt?

File: src/node/element/path/data.rs [hunk 5]
```diff
@@ -250,10 +249,10 @@ impl<'l> Parser<'l> {
 
 #[cfg(test)]
 mod tests {
-    use node::Value;
-    use super::{Data, Parser};
     use super::super::Command::*;
     use super::super::Position::*;
+    use super::{Data, Parser};
+    use crate::node::Value;
```
Comment (IvanUkhov) [inline src/node/element/path/data.rs:255]: Is it by rustfmt?

**Review by IvanUkhov — COMMENTED** [review]
Thank you for your time! Can you please motivate the need for 2018 in brief?

File: src/node/element/mod.rs (hunk 2 above)
Comment (octronics) [reply src/node/element/mod.rs:253]: No, It is by `cargo fix --edition` on commit d6159f4ef2c8fce81103cbc7ecabc4a0b01270b5

File: src/node/element/path/data.rs (hunk 3 above)
Comment (octronics) [reply src/node/element/path/data.rs:5]: No, It is by `cargo fix --edition` on commit d6159f4ef2c8fce81103cbc7ecabc4a0b01270b5

File: src/node/element/path/data.rs (hunk 5 above)
Comment (octronics) [reply src/node/element/path/data.rs:255]: No, It is by `cargo fix --edition` on commit d6159f4ef2c8fce81103cbc7ecabc4a0b01270b5

File: src/parser/reader.rs (hunk 4 above)
Comment (octronics) [reply src/parser/reader.rs:119]: Yes, commit 858c43eb304084832222ba615c02bffdb4810ffc

**Comment by IvanUkhov** [comment]
Thank you!

**Comment by IvanUkhov** [comment]
@octronics, how would you like to be added to the authors list? Should it be `OCTronics <octronics@riseup.net>`, or would you like your actual name to be there?

**Comment by octronics** [comment]
Please use OCTronics for attribution. Thank you for asking.

**Pull request merged**